packages/react-data-grid/src/ReactDataGrid.tsx

```diff
@@ -0,0 +1,729 @@
+   * Callback called whenever row data is updated
+   * When editing is enabled, this callback will be called for the following scenarios
+   * 1. Using the supplied editor of the column. The default editor is the [SimpleTextEditor](https://github.com/adazzle/react-data-grid/blob/master/packages/common/editors/SimpleTextEditor.js).
+   * 2. Copy/pasting the value from one cell to another <kbd>CTRL</kbd>+<kbd>C</kbd>, <kbd>CTRL</kbd>+<kbd>V</kbd>
+   * 3. Update multiple cells by dragging the fill handle of a cell up or down to a destination cell.
+   * 4. Update all cells under a given cell by double clicking the cell's fill handle.
+   */
+  onGridRowsUpdated?(args: {
+    cellKey: string;
+    fromRow: number;
+    toRow: number;
+    fromRowId: unknown;
+    toRowId: unknown;
+    rowIds: unknown[];
+    updated: { [key: string]: unknown };
+    action: UpdateActions;
+    fromRowData: RowData;
+  }): void;
+}
+
+interface State {
+  columnMetrics: ColumnMetrics;
+  lastRowIdxUiSelected: number;
+  selectedRows: RowData[];
+  canFilter?: boolean;
+  sortColumn?: string;
+  sortDirection?: DEFINE_SORT;
+}
+
+function isRowSelected(keys: unknown, indexes: unknown, isSelectedKey: unknown, rowData: RowData, rowIdx: number) {
+  return rowUtils.isRowSelected(keys as { rowKey?: string; values?: string[] } | null, indexes as number[] | null, isSelectedKey as string | null, rowData, rowIdx);
+}
+
+/** Main API Component to render a data grid of rows and columns
+ *
+ * Example code
+ * -----
+ *
+ * ```javascript
+ * <ReactDataGrid
+ *   columns={columns}
+ *   rowGetter={i => rows[i]}
+ *   rowsCount={3} />
+ * ```
+*/
+export default class ReactDataGrid extends React.Component<Props, State> {
+  static displayName = 'ReactDataGrid';
+
+  static defaultProps = {
+    enableCellSelect: false,
+    rowHeight: 35,
+    headerFiltersHeight: 45,
+    enableRowSelect: false,
+    minHeight: 350,
+    rowKey: 'id',
+    cellNavigationMode: CellNavigationMode.NONE,
+    enableCellAutoFocus: true,
+    minColumnWidth: 80,
+    columnEquality: sameColumn,
+    editorPortalTarget: document.body
+  };
+
+  private readonly grid = React.createRef<HTMLDivElement>();
+  private readonly base = React.createRef<Grid>();
+  private readonly selectAllCheckbox = React.createRef<HTMLInputElement>();
+  private readonly eventBus = new EventBus();
+  private readonly _keysDown: { [key: number]: boolean } = {};
+  private _cachedColumns?: Column[];
+  private _cachedComputedColumns?: ColumnList;
+
+  readonly state: Readonly<State>;
+
+  constructor(props: Props) {
+    super(props);
+    const initialState: State = {
+      columnMetrics: this.createColumnMetrics(),
+      selectedRows: [],
+      canFilter: false,
+      lastRowIdxUiSelected: -1
+    };
+
+    if (this.props.sortColumn && this.props.sortDirection) {
+      initialState.sortColumn = this.props.sortColumn;
+      initialState.sortDirection = this.props.sortDirection;
+    }
+    this.state = initialState;
```

Comment:
Need to use constructor to initialize state as it relies on some methods that should be initialized before state is set